File: workspaces-good/packages/shared/src/taxonomy_operation_vo.ts
export enum OperationVO {
  Add = "+",
  Subtract = "-",
  Multiply = "*",
  Divide = "/",
}

export function operationFromSymbol(s: string): OperationVO | null {
  switch (s) {
    case "+": return OperationVO.Add;
    case "-": return OperationVO.Subtract;
    case "*":
    case "x": return OperationVO.Multiply;
    case "/": return OperationVO.Divide;
    default: return null;
  }
}

export function operationSymbol(op: OperationVO): string {
  return op;
}

File: workspaces-good/modules/shared/src/taxonomy_operation_vo.py
from enum import Enum


class OperationVO(Enum):
    ADD = "+"
    SUBTRACT = "-"
    MULTIPLY = "*"
    DIVIDE = "/"

    def symbol(self) -> str:
        return self.value


def operation_from_symbol(s: str) -> "OperationVO | None":
    mapping = {
        "+": OperationVO.ADD,
        "-": OperationVO.SUBTRACT,
        "*": OperationVO.MULTIPLY,
        "x": OperationVO.MULTIPLY,
        "/": OperationVO.DIVIDE,
    }
    return mapping.get(s)
